mne/tests/test_cov.py

```diff
@@ -496,6 +497,10 @@ def test_cov_scaling():
     _undo_scaling_array(data, picks_list, scalings=scalings)
     assert_allclose(data, evoked.data, atol=1e-20)
 
+    # check that input data remain unchanged. gh-5698
+    _regularized_covariance(data)
+    assert_array_almost_equal(data, evoked.data)
```

Comment:
@DavidSabbagh can you confirm that this line is failing on master? Does the test crash if you comment out the covariance unscaling line ?

Reply:
> can you confirm that this line is failing on master?

It does

```sh
~/code/mne-python DavidSabbagh-fix_cov_scaling*
(mne) ❯ git pull                                 
From github.com:DavidSabbagh/mne-python
 * branch                fix_cov_scaling -> FETCH_HEAD
Updating 750e6a102..33d331a6e
Fast-forward
 doc/whats_new.rst                           |  4 ++++
 examples/decoding/plot_decoding_spoc_CMC.py |  6 +++---
 mne/tests/test_cov.py                       | 13 +++----------
 3 files changed, 10 insertions(+), 13 deletions(-)

~/code/mne-python DavidSabbagh-fix_cov_scaling*
(mne) ❯ pytest mne/tests/test_cov.py -k scaling
Test session starts (platform: linux, Python 3.6.6, pytest 3.6.3, pytest-sugar 0.9.1)
rootdir: /home/sik/code/mne-python, inifile: setup.cfg
plugins: smother-0.2, sugar-0.9.1, pudb-0.6, ipdb-0.1, faulthandler-1.5.0, cov-2.5.1

 mne/tests/test_cov.py ✓                                                                  100% ██████████
======================================= slowest 20 test durations =======================================
0.20s call     mne/tests/test_cov.py::test_cov_scaling
0.00s setup    mne/tests/test_cov.py::test_cov_scaling
0.00s teardown mne/tests/test_cov.py::test_cov_scaling

Results (0.43s):
       1 passed
      14 deselected

~/code/mne-python DavidSabbagh-fix_cov_scaling*
(mne) ❯ git checkout master                      
Switched to branch 'master'
Your branch is up to date with 'mne/master'.

~/code/mne-python master*
(mne) ❯ git checkout DavidSabbagh-fix_cov_scaling mne/tests/test_cov.py 

~/code/mne-python master*
(mne) ❯ pytest mne/tests/test_cov.py -k scaling                        
Test session starts (platform: linux, Python 3.6.6, pytest 3.6.3, pytest-sugar 0.9.1)
rootdir: /home/sik/code/mne-python, inifile: setup.cfg
plugins: smother-0.2, sugar-0.9.1, pudb-0.6, ipdb-0.1, faulthandler-1.5.0, cov-2.5.1


――――――――――――――――――――――――――――――――――――――――――― test_cov_scaling ――――――――――――――――――――――――――――――――――――――――――――

    def test_cov_scaling():
        """Test rescaling covs."""
        evoked = read_evokeds(ave_fname, condition=0, baseline=(None, 0),
                              proj=True)
        cov = read_cov(cov_fname)['data']
        cov2 = read_cov(cov_fname)['data']
    
        assert_array_equal(cov, cov2)
        evoked.pick_channels([evoked.ch_names[k] for k in pick_types(
            evoked.info, meg=True, eeg=True
        )])
        picks_list = _picks_by_type(evoked.info)
        scalings = dict(mag=1e15, grad=1e13, eeg=1e6)
    
        _apply_scaling_cov(cov2, picks_list, scalings=scalings)
        _apply_scaling_cov(cov, picks_list, scalings=scalings)
        assert_array_equal(cov, cov2)
        assert cov.max() > 1
    
        _undo_scaling_cov(cov2, picks_list, scalings=scalings)
        _undo_scaling_cov(cov, picks_list, scalings=scalings)
        assert_array_equal(cov, cov2)
        assert cov.max() < 1
    
        data = evoked.data.copy()
        _apply_scaling_array(data, picks_list, scalings=scalings)
        _undo_scaling_array(data, picks_list, scalings=scalings)
        assert_allclose(data, evoked.data, atol=1e-20)
    
        # check that input data remain unchanged. gh-5698
        _regularized_covariance(data)
>       assert_array_almost_equal(data, evoked.data)
E       AssertionError: 
E       Arrays are not almost equal to 6 decimals
E       
E       (mismatch 16.6913282193061%)
E        x: array([-6.375903e-08,  9.696686e-08,  3.862735e-07, ...,  1.098398e-01,
E               1.877830e-01,  2.230980e-01])
E        y: array([-6.375903e-13,  9.696686e-13,  3.862735e-12, ...,  1.098398e-06,
E               1.877830e-06,  2.230980e-06])

cov        = array([[ 3.50127927e-23,  6.87212230e-24,  9.42337323e-25, ...,
         4.31433961e-18,  1.59003769e-18,  8.71581032e...  [ 8.71581032e-18,  4.47960458e-18,  3.59028919e-19, ...,
         1.31563867e-11,  1.05242517e-11,  1.90726354e-11]])
cov2       = array([[ 3.50127927e-23,  6.87212230e-24,  9.42337323e-25, ...,
         4.31433961e-18,  1.59003769e-18,  8.71581032e...  [ 8.71581032e-18,  4.47960458e-18,  3.59028919e-19, ...,
         1.31563867e-11,  1.05242517e-11,  1.90726354e-11]])
data       = array([[-6.37590281e-08,  9.69668608e-08,  3.86273487e-07, ...,
         6.43434942e-07,  2.57692775e-07, -6.37590281e...  [-1.69842720e-02, -6.42192410e-02, -1.21045207e-01, ...,
         1.09839781e-01,  1.87783030e-01,  2.23097985e-01]])
evoked     = <Evoked  |  'Left Auditory' (mean, N=3), [-0.1998, 0.49949] sec, 366 ch, ~4.8 MB>
picks_list = [('mag', array([  2,   5,   8,  11,  14,  17,  20,  23,  26,  29,  32,  35,  38,
        41,  44,  47,  50,  53,  56, ...     345, 346, 347, 348, 349, 350, 351, 352, 353, 354, 355, 356, 357,
       358, 359, 360, 361, 362, 363, 364, 365]))]
scalings   = {'eeg': 1000000.0, 'grad': 10000000000000.0, 'mag': 1000000000000000.0}

mne/tests/test_cov.py:502: AssertionError

 mne/tests/test_cov.py ⨯                                                                  100% ██████████
======================================= slowest 20 test durations =======================================
0.19s call     mne/tests/test_cov.py::test_cov_scaling
0.00s setup    mne/tests/test_cov.py::test_cov_scaling
0.00s teardown mne/tests/test_cov.py::test_cov_scaling

Results (0.61s):
       1 failed
         - mne/tests/test_cov.py:471 test_cov_scaling
      14 deselected

~/code/mne-python master*
(mne) ❯ git reset --hard
HEAD is now at ae38e1b79 [ENH] [MRG+1] Unify read annotations (#5695)
```

> Does the test crash if you comment out the covariance unscaling line ?

I've no idea. @DavidSabbagh ?